c/src/util/small_storage.cpp

```diff
@@ -0,0 +1,36 @@
+#include <cstring>
+#include <memory>
+
+// copy_or_allocate_into_aligned_storage is a helper that will allocate space to hold an object and some aligned data
+// after it in one allocation. A return value of 0 indicates that the buffer pointed to by 'space' was sufficient for
+// holding the additional data. a non-zero return means a buffer was allocated large enough to include the
+// minimum size as determined by the offset.
+void* copy_or_allocate_into_aligned_storage(
```

Comment:
suggestion: the large stateful op test is failing. Let's a standalone test for the small storage

Reply:
Will save unit tests for later.